optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -625,4 +643,231 @@ private boolean shouldIgnoreMember(ClassInfo declaringClass) {
         return declaringClass.name().toString().startsWith(SolutionDescriptor.class.getName());
     }
 
+    private void registerCustomClassesFromSolverConfig(SolverConfig solverConfig, Set<Class<?>> reflectiveClassSet) {
+        ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = solverConfig.getScoreDirectorFactoryConfig();
+        while (scoreDirectorFactoryConfig != null) {
+            registerClassForReflection(scoreDirectorFactoryConfig.getEasyScoreCalculatorClass(), reflectiveClassSet);
+            registerClassForReflection(scoreDirectorFactoryConfig.getConstraintProviderClass(), reflectiveClassSet);
+            registerClassForReflection(scoreDirectorFactoryConfig.getIncrementalScoreCalculatorClass(), reflectiveClassSet);
+            scoreDirectorFactoryConfig = scoreDirectorFactoryConfig.getAssertionScoreDirectorFactory();
```

Comment:
I believe there's a fail fast if a non-null getAssertionScoreDirectorFactory has itself a non-null getAssertionScoreDirectorFactory, so this can just be an if instead of a while.